### ISSUE TITLE
Bug: Unions broken in python 3.10 on dataclass_utils.into

### ISSUE THREAD

**Issue by llucid-97** [tamuhey/dataclass_utils#9]
Hello, I ran into a bug similar to https://github.com/tamuhey/dataclass_utils/issues/8
If i try using the python 3.10 union syntax sugar on dataclass type hints then try to call dataclass_utils.into on them, it breaks, but doing so with a normal typing union doesn't even though they should be equivalent

```python
import dataclasses
import dataclass_utils
import typing

print(int | str)
print(typing.Union[int, str])

@dataclasses.dataclass
class FooUnion:
    foo: typing.Union[str, int] = "foo"

@dataclasses.dataclass
class FooSugar:
    foo: str | int = "foo"

f_union = FooUnion()
f_dict = dataclasses.asdict(f_union)
f_union_from_dict = dataclass_utils.into(f_dict, FooUnion)
print(f"Pass! {f_union_from_dict=}")

f_sugar = FooSugar()
f_dict = dataclasses.asdict(f_sugar)
f_sugar_from_dict = dataclass_utils.into(f_dict, FooSugar)
print(f"Pass! {f_sugar_from_dict=}")
```

> Traceback (most recent call last):
>   File "[...]\dataclass_utils_bug.py", line 23, in <module>
>     f_sugar_from_dict = dataclass_utils.into(f_dict, FooSugar)
>   File "[...]\lib\site-packages\dataclass_utils\into_dataclass.py", line 24, in into_root
>     raise ret
> dataclass_utils.error.Error0: Error in field 'foo'. Expected type str | int, got <class 'str'> (value: foo)
> int | str
> typing.Union[int, str]
> Pass! f_union_from_dict=FooUnion(foo='foo')
> 
> Process finished with exit code 1

**Comment by tamuhey** [comment]
Thanks for reporting the bug! I'll investigate it.

**Comment by tamuhey** [comment]
Just released version 0.7.23, which should fix this issue.
